pkg/shell/base_index.js

```diff
@@ -481,6 +478,20 @@ function Router(index) {
 
             /* Add the child's group to all open channel messages */
             } else if (control.command == "open") {
+                // Prevent frames from malicious remote hosts from tampering with other machines
+                // The lookup may fail if a frame currently being removed
+                const child_host = source_by_name[child.name]?.default_host;
+                if (child_host && child_host !== "localhost" && child_host !== control.host) {
```

Comment:
Why not also apply this restriction for `localhost`?

Reply:
We still do want the frames from the primary login machine to access other machines, like RHCERT. That feels like a safe enough use case, at least it's outside of "connecting to remote untrusted SSH machines can backfire to the client".